src/Workspaces/Remote/Core/SolutionAssetProvider.cs

```diff
@@ -91,57 +91,39 @@ private async ValueTask GetAssetsWorkerAsync(PipeWriter pipeWriter, Checksum sol
         /// <em>every</em> write.  That's undesirable as that will then block a thread pool thread on the actual
         /// asynchronous flush call to the underlying PipeWriter
         /// </summary>
+        /// <remarks>
+        /// Note: this stream does not have to <see cref="PipeWriter.Complete"/> the underlying <see cref="_writer"/> it
+        /// is holding onto (including within <see cref="Flush"/>, <see cref="FlushAsync"/>, or <see cref="Dispose"/>).
+        /// Responsibility for that is solely in the hands of <see cref="GetAssetsAsync"/>.
+        /// </remarks>
         private class PipeWriterStream : Stream, IDisposableObservable
         {
             private readonly PipeWriter _writer;
 
-            internal PipeWriterStream(PipeWriter writer)
-            {
-                _writer = writer ?? throw new ArgumentNullException(nameof(writer));
-            }
+            public bool IsDisposed { get; private set; }
```

Comment:
this change looks messy, but it was asll cleanup/moves.

1. properties moved up.
2.  constructor removed argumentnullcheck
3. dispose moved under contructor
4. "read helpers" region moved below everything as that code is not interesting. 

functionality-wise, this is all the same.